### PR TITLE
Tycho: Allows ids for tags

### PR DESCRIPTION
**Pull request by cschierle**
Fixes: [SIRI-606](https://scireum.myjetbrains.com/youtrack/issue/SIRI-606)

### CODE REVIEW

File: src/main/resources/default/taglib/t/heading.html.pasta [hunk 1]
```diff
@@ -1,17 +1,17 @@
 <i:arg type="String" name="labelKey" default=""/>
 <i:arg type="String" name="label" default="@i18n(labelKey)"/>
-<i:arg type="String" name="class" default="" />
-<i:local name="markupLabel" value="@renderToString('body')" />
+<i:arg type="String" name="class" default=""/>
+<i:local name="markupLabel" value="@renderToString('body')"/>
 <i:if test="isFilled(label)">
```
Comment (jakobvogel) [inline src/main/resources/default/taglib/t/heading.html.pasta:5]: I personally prefer some blank lines here, but that is – of course – a matter of taste.

```suggestion
<i:arg type="String" name="class" default=""/>

<i:local name="markupLabel" value="@renderToString('body')"/>

<i:if test="isFilled(label)">
```

File: src/main/resources/default/taglib/t/heading.html.pasta [hunk 2]
```diff
@@ -1,17 +1,17 @@
...
     <div class="bb-gray pb-2 mt-4 mb-3 h5 text-black-75 align-items-end d-flex flex-row @class">
-        <span class="overflow-hidden">@label</span>
-        <span class="ml-auto"><i:render name="actions" /></span>
+        <span class="overflow-hidden">@label<i:render name="label"/></span>
```
Comment (jakobvogel) [inline src/main/resources/default/taglib/t/heading.html.pasta:7]: Do we really want to support both? In similar cases, we often seem to check for whether one or the other is filled and only print one.

File: src/main/resources/default/taglib/t/heading.html.pasta [hunk 3]
```diff
@@ -1,17 +1,17 @@
...
+        <span class="ml-auto"><i:render name="actions"/></span>
     </div>
     <i:else>
         <i:if test="isFilled(markupLabel)">
             <div class="bb-gray pb-2 mt-4 mb-3 h5 text-black-75 align-items-end d-flex flex-row @class">
-                <span class="overflow-hidden"><i:raw>@markupLabel</i:raw></span>
-                <span class="ml-auto"><i:render name="actions" /></span>
+                <span class="overflow-hidden"><i:raw>@markupLabel<i:render name="label"/></i:raw></span>
```
Comment (jakobvogel) [inline src/main/resources/default/taglib/t/heading.html.pasta:13]: (s.a.)

File: src/main/resources/default/taglib/t/tag.html.pasta [hunk 4]
```diff
@@ -4,14 +4,15 @@
        default="true"
        description="Determines if the dot has an auto margin at the right"/>
 <i:arg type="String" name="class" default="" description="Defines addition classes to add."/>
+<i:arg type="String" name="id" default="" description="Defines the id of the rendered span."/>
```
Comment (jakobvogel) [inline src/main/resources/default/taglib/t/tag.html.pasta:7]: Let's try not to unnerve @sabieber 😉

```suggestion
<i:arg type="String" name="id" default="" description="Defines the ID of the rendered span."/>
```

File: src/main/resources/default/taglib/t/heading.html.pasta (hunk 2 above)
Comment (cschierle) [reply src/main/resources/default/taglib/t/heading.html.pasta:9]: The use case is to place additional information next to a heading provided by a sirius lib via an extension in a product. This seemed to be the easiest way to achieve this.

Alternatively, I could also try and find another method to append things via JS.

Feel free to comment, which approach I should rather follow.

Comment (jakobvogel) [reply src/main/resources/default/taglib/t/heading.html.pasta:9]: That sounds legit. Then I am just wondering whether the block should be named `label-extension` instead.

In the straight-forward use case, I find it a bit strange to do the content as attribute and the extension as nested block in this case. But that is also a matter of taste, of course.

Comment (jakobvogel) [reply src/main/resources/default/taglib/t/heading.html.pasta:9]: (Add yes, I would hope that people would use the body version)

**Review by sabieber — APPROVED** [review]

**Review by idlira — APPROVED** [review]

File: src/main/resources/default/taglib/t/heading.html.pasta (hunk 2 above)
Comment (andyHa) [reply src/main/resources/default/taglib/t/heading.html.pasta:9]: commonly the i:block replaces the parameter. Check card.html.pasta - we ofter renderToString as fallback to the plain parameter..I'd keep it consistent here

**Review by andyHa — CHANGES_REQUESTED** [review]
We should probably stay consistent here...

File: src/main/resources/default/taglib/t/heading.html.pasta (hunk 2 above)
Comment (jakobvogel) [reply src/main/resources/default/taglib/t/heading.html.pasta:9]: For the consistent case, there is already `markupLabel`. @cschierle's use case is truly remarkable 😅

Comment (andyHa) [reply src/main/resources/default/taglib/t/heading.html.pasta:9]: Where do we use this at all?

**Review by sabieber — APPROVED** [review]

**Review by idlira — APPROVED** [review]

**Review by jakobvogel — APPROVED** [review]

**Pull request merged**